rviz_common/include/rviz_common/message_filter_display.hpp

```diff
@@ -103,6 +103,27 @@ class MessageFilterDisplay : public _RosTopicDisplay
 protected:
   void updateTopic() override
   {
+    // Change the QoS according to the info in the ROS Graph
+    const auto & publishers = rviz_ros_node_.lock()->get_raw_node()->get_publishers_info_by_topic(
+      topic_property_->getTopicStd());
+
+    if (!publishers.empty()) {
+      // Check if there is any Subscriber with the same QoS as set
+      bool any_publisher_qos_compatible = false;
+      for (const auto & publisher : publishers) {
+        if (qos_profile == publisher.qos_profile()) {
```

Comment:
The profile doesn't have to be the same in order to be compatible.
We should check for compatibility instead.

There's https://github.com/ros2/rmw/blob/05f973575e8e93454e39f51da6227509061ff189/rmw/include/rmw/qos_profiles.h#L176.

There's no rclcpp wrapper, so you can either use the rmw function directly or add one.